packages/components/src/tree/tree.tsx

```diff
@@ -39,26 +38,48 @@ export const {
     expandedItems: string[];
 }>(defaultRoot);
 
-export interface TreeAddedProps<T> {
+export type GetChildren<T> = (t: T) => T[];
+export type GetId<T> = (t: T) => string;
+
+export interface TreeAddedProps<T, EL extends HTMLElement> {
     data: T;
-    getChildren: (t: T) => T[];
+    getChildren: GetChildren<T>;
     openItemsControls: StateControls<string[]>;
-
+    eventsRoot?: React.RefObject<EL>;
     ItemRenderer: React.ComponentType<TreeItemProps<T>>;
-
     overlay?: typeof overlayRoot;
 }
 
 export type TreeProps<T, EL extends HTMLElement> = Omit<
     ScrollListProps<T, EL, TreeItemInfo<T>>,
     'items' | 'ItemRenderer'
 > &
-    TreeAddedProps<T>;
+    TreeAddedProps<T, EL>;
+
+export const getAllTreeItems = <T,>({
+    item,
+    getChildren,
+    getId,
+}: {
+    item: T;
+    getChildren: GetChildren<T>;
+    getId: GetId<T>;
+}): T[] => [
```

Comment:
would look more clean imo if you clear an interface for the params, just a personal preference though.